### PR TITLE
adding data agent queue

### PR DESCRIPTION
**Pull request by exocom**
## All Submissions:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] I have updated the documentation accordingly.
* [X] I have added tests to cover my changes.
* [X] All new and existing tests passed.
* [X] My code follows the code style of this project.
* [X] I ran lint checks locally prior to submission.
* [X] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Adding data source for Agent Queues.

Issue Number:

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [X] No

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by exocom** [comment]
@xuzhang3 can you help me with this build failure?

**Comment by xuzhang3** [comment]
> @xuzhang3 can you help me with this build failure?

Dependency package names has been changed from `terraform-providers` to `microsoft`. The build checks should be working now.

**Comment by exocom** [comment]
> > @xuzhang3 can you help me with this build failure?
> 
> Dependency package names has been changed from `terraform-providers` to `microsoft`. The build checks should be working now.

Thank you!!! That had me stumped.

File: azuredevops/internal/service/taskagent/data_agent_queue.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		Project:   projectID,
+		QueueName: name,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(*agentQueues) > 1 {
+		return nil, fmt.Errorf("Found multiple agent queues for name: %s. Agent queues found: %v", *name, agentQueues)
```
Comment (xuzhang3) [inline azuredevops/internal/service/taskagent/data_agent_queue.go]: Suggest to use `%+v` to print out the param names

File: azuredevops/internal/service/taskagent/data_agent_queue.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	flattenAzureAgentQueue(d, agentQueue)
+	return nil
+}
+
+func flattenAzureAgentQueue(d *schema.ResourceData, agentQueue *taskagent.TaskAgentQueue) {
+	d.SetId(strconv.Itoa(*agentQueue.Id))
+	d.Set("name", converter.ToString(agentQueue.Name, ""))
+	d.Set("pool_id", strconv.Itoa(*agentQueue.Pool.Id))
```
Comment (xuzhang3) [inline azuredevops/internal/service/taskagent/data_agent_queue.go]: `pool_id` is not defined in the schema

**Review by xuzhang3 — COMMENTED** [review]
`data_agent_queue.html.markdown` should add to `azuredevops.erb`, otherwise the document won't be displayed in  https://www.terraform.io/docs.

**Comment by exocom** [comment]
@xuzhang3 can you let me know what are the build failures?

**Comment by xuzhang3** [comment]
@exocom 
```log
2020/09/10 11:54:38 [WARN] Provider "azuredevops" produced an unexpected new value for azuredevops_project.project, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .version_control: was cty.StringVal("git"), but now cty.StringVal("Git")
      - .description: was null, but now cty.StringVal("")
panic: project_id: '' expected type 'string', got unconvertible type 'uuid.UUID'

goroutine 315 [running]:
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).Set(0xc0003ca770, 0x14c4970, 0xa, 0x142dbe0, 0xc000581490, 0x0, 0x0)
	/home/vsts/work/1/s/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource_data.go:201 +0x354
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/taskagent.flattenAzureAgentQueue(0xc0003ca770, 0xc000116d00)
	/home/vsts/work/1/s/azuredevops/internal/service/taskagent/data_agent_queue.go:56 +0x16c
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/taskagent.dataSourceAgentQueueRead(0xc0003ca770, 0x120c420, 0xc000110600, 0xc0003ca770, 0x0)
	/home/vsts/work/1/s/azuredevops/internal/service/taskagent/data_agent_queue.go:48 +0x1d2
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc000491320, 0xc000598a40, 0x120c420, 0xc000110600, 0xc00043fc30, 0x1, 0x0)
	/home/vsts/work/1/s/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:403 +0x88
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc000116b00, 0xc000503a88, 0xc000598a40, 0xc000598a40, 0x0, 0x0)
	/home/vsts/work/1/s/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:451 +0x8f
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc000010b50, 0x1802320, 0xc0003cdb00, 0xc000162f80, 0xc000010b50, 0xc0003cdb00, 0xc000013b30)
	/home/vsts/work/1/s/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:1036 +0x44e
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler(0x1466440, 0xc000010b50, 0x1802320, 0xc0003cdb00, 0xc0000c0ea0, 0x0, 0x1802320, 0xc0003cdb00, 0xc0001a2400, 0x77)
	/home/vsts/work/1/s/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3341 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004ef680, 0x1815a60, 0xc0004ef800, 0xc000463400, 0xc000ab2600, 0x210a8d0, 0x0, 0x0, 0x0)
	/home/vsts/work/1/s/vendor/google.golang.org/grpc/server.go:1024 +0x4f4
google.golang.org/grpc.(*Server).handleStream(0xc0004ef680, 0x1815a60, 0xc0004ef800, 0xc000463400, 0x0)
	/home/vsts/work/1/s/vendor/google.golang.org/grpc/server.go:1313 +0xd97
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000580034, 0xc0004ef680, 0x1815a60, 0xc0004ef800, 0xc000463400)
	/home/vsts/work/1/s/vendor/google.golang.org/grpc/server.go:722 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/vsts/work/1/s/vendor/google.golang.org/grpc/server.go:720 +0xa1
FAIL	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests	28.585s
```

**Comment by xuzhang3** [comment]
I have changed the project visibility  to public, you should be able to see the pipeline build  details now.

**Comment by exocom** [comment]
@xuzhang3 Solved it. Error was caused by implicit conversion of  uuid to string. My local version of go was on 1.15 so it worked for me.

Thanks for making the pipeline visible that is a huge help!

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**